pkg/apis/serving/v1beta1/inference_service_validation.go

```diff
@@ -141,6 +147,57 @@ func validateInferenceService(isvc *InferenceService) (admission.Warnings, error
 	return allWarnings, nil
 }
 
+// validateMultiNodeVariables validates when there is workerSpec set in isvc
+func validateMultiNodeVariables(isvc *InferenceService) error {
+	if isvc.Spec.Predictor.WorkerSpec != nil {
+		if isvc.Spec.Predictor.Model != nil {
+			if _, exists := utils.GetEnvVarValue(isvc.Spec.Predictor.Model.PredictorExtensionSpec.Container.Env, constants.PipelineParallelSizeEnvName); exists {
+				return fmt.Errorf(DisallowedWorkerSpecPipelineParallelSizeEnvError, isvc.Name)
+			}
+			if _, exists := utils.GetEnvVarValue(isvc.Spec.Predictor.Model.PredictorExtensionSpec.Container.Env, constants.TensorParallelSizeEnvName); exists {
+				return fmt.Errorf(DisallowedWorkerSpecTensorParallelSizeEnvError, isvc.Name)
+			}
+			if utils.IsUnknownGpuResourceType(isvc.Spec.Predictor.Model.Resources) {
+				return fmt.Errorf(InvalidUnknownGPUTypeError, isvc.Name)
+			}
+			if isvc.Spec.Predictor.Model.StorageURI == nil {
+				return fmt.Errorf(MissingStorageURI, isvc.Name)
+			} else {
+				storageProtocol := strings.Split(*isvc.Spec.Predictor.Model.StorageURI, "://")[0]
+				if storageProtocol != "pvc" {
+					return fmt.Errorf(InvalidNotSupportedStorageURIProtocolError, isvc.Name, storageProtocol)
+				}
+			}
+			if isvc.GetAnnotations()[constants.AutoscalerClass] != string(constants.AutoscalerClassExternal) {
+				return fmt.Errorf(InvalidAutoScalerError, isvc.Name, isvc.GetAnnotations()[constants.AutoscalerClass])
+			}
+		}
+
+		// WorkerSpec.PipelineParallelSize should not be less than 2 (head + worker)
+		if isvc.Spec.Predictor.WorkerSpec.PipelineParallelSize != nil {
+			workerSpecPipelineParallelSize := *isvc.Spec.Predictor.WorkerSpec.PipelineParallelSize
+			if workerSpecPipelineParallelSize < 2 {
+				return fmt.Errorf(InvalidWorkerSpecPipelineParallelSizeValueError, isvc.Name, strconv.Itoa(workerSpecPipelineParallelSize))
+			}
+		}
+		if isvc.Spec.Predictor.WorkerSpec.TensorParallelSize != nil {
+			// WorkerSpec.TensorParallelSize should not be less than 1.
+			workerSpecTensorParallelSize := *isvc.Spec.Predictor.WorkerSpec.TensorParallelSize
+			if workerSpecTensorParallelSize < 1 {
+				return fmt.Errorf(InvalidWorkerSpecTensorParallelSizeValueError, isvc.Name, strconv.Itoa(workerSpecTensorParallelSize))
+			}
+		}
+		if isvc.Spec.Predictor.WorkerSpec.Containers != nil {
+			for _, container := range isvc.Spec.Predictor.WorkerSpec.Containers {
```

Comment:
Under the concept of workerSpec, setting up multiple containers is meaningless. However, I think it's acceptable to inherit the existing PodSpec that includes containers. Therefore, I believe it would be a good idea to add validation to check if multiple containers have been set.